lib/executor/execution_test.go

```diff
@@ -39,24 +40,62 @@ import (
 
 func TestExecutionStateVUIDs(t *testing.T) {
 	t.Parallel()
-	et, err := lib.NewExecutionTuple(nil, nil)
-	require.NoError(t, err)
-	es := lib.NewExecutionState(lib.Options{}, et, 0, 0)
-	assert.Equal(t, uint64(1), es.GetUniqueVUIdentifier())
-	assert.Equal(t, uint64(2), es.GetUniqueVUIdentifier())
-	assert.Equal(t, uint64(3), es.GetUniqueVUIdentifier())
-	wg := sync.WaitGroup{}
-	rand.Seed(time.Now().UnixNano())
-	count := 100 + rand.Intn(50)
-	wg.Add(count)
-	for i := 0; i < count; i++ {
-		go func() {
-			es.GetUniqueVUIdentifier()
-			wg.Done()
-		}()
+
+	testCases := []struct {
+		seq, seg string
+	}{
+		{},
+		{seq: "0,1/4,3/4,1", seg: "0:1/4"},
+		{seq: "0,0.3,0.5,0.6,0.7,0.8,0.9,1", seg: "0.5:0.6"},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(fmt.Sprintf("seq:%s;segment:%s", tc.seq, tc.seg), func(t *testing.T) {
+			t.Parallel()
+			ess, err := lib.NewExecutionSegmentSequenceFromString(tc.seq)
+			require.NoError(t, err)
+			segment, err := lib.NewExecutionSegmentFromString(tc.seg)
+			require.NoError(t, err)
+			et, err := lib.NewExecutionTuple(segment, &ess)
+			require.NoError(t, err)
+
+			start, offsets, _ := et.GetStripedOffsets()
+			es := lib.NewExecutionState(lib.Options{}, et, 0, 0)
+
+			idl, idg := es.GetUniqueVUIdentifiers()
+			assert.Equal(t, uint64(1), idl)
+			expGlobal := start + 1
+			assert.Equal(t, uint64(expGlobal), idg)
+
+			idl, idg = es.GetUniqueVUIdentifiers()
+			assert.Equal(t, uint64(2), idl)
+			expGlobal += offsets[0]
+			assert.Equal(t, uint64(expGlobal), idg)
+
+			idl, idg = es.GetUniqueVUIdentifiers()
+			assert.Equal(t, uint64(3), idl)
+			expGlobal += offsets[0]
```

Comment:
Shouldn't this be `offsets[1]` ? 

Also I feel like this is basically testing that it uses execution segments underneed, which doesn't seem that useful.

A concrete example with concrete numbers at least means that it requires that "given this segmented execution parameters we will be getting this VU ids". In the case of executors this was mostly done because 
1. there is a lot more logic *based* on the execution segments
2. sometimes some of the numbers are *not* based on the original execution segments provided as configuration, but based on something on top of them for the concrete scenario and executor. 

What I am trying to say is that this IMO doesn't test anything new, apart from "GetUniqueVUIdentifiers uses segemetned index underneed" which seems like an implementation detail to me(albeit the one that makes this possible at all)

Reply:
`offsets` has a single element for these execution segments, and I picked them more or less deliberately so that I wouldn't have to cycle through multiple offsets :sweat_smile:

You're right that this is more of a white-box test now, but we have plenty of those around... Arguably the previous test was also white-box, it's just that the implementation was to always increment by 1, so it was simpler to check.

I wanted to keep this same test style so that I wouldn't have to build and assert against `expectedIDs`, as it would be unwieldy, and I like the current random approach.

How would you rewrite it or test this differently?

Reply:
1. I probably will just *not* test it.
2.  Or just expect that the IDs will go up
2+. Check that with all the different segments of a given sequence I will *always* get different global VU ids between the segments. Possibly that there are no holes as well

`2.` in my opinion test the important property of the whole thing which is that we will have *different* global VU IDs between the instances, not what their exact numbers will be.

Reply:
*but* IMO we can do this later as we need to finish up the main PR so ... merge this as is :P